### PR TITLE
Add Dockerfile for PanGenie

### PR DESCRIPTION
**Pull request by VJalili**
This PR adds a Dockerfile for [PanGenie](https://bitbucket.org/jana_ebler/pangenie/src/master/).

### CODE REVIEW

File: dockerfiles/pangenie/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#   RUN bash miniconda.sh -b -p /opt/conda && \
+#       /opt/conda/bin/conda init bash && \
+#       bash -c "source ${HOME}/.bashrc"
+#
+# A similar issue is detailed in the following blog post; therefore, as
+# suggested in this blog post, this docker files is using
+# 'continuumio/miniconda3' as base image.
+#   https://pythonspeed.com/articles/activate-conda-dockerfile/
+
+FROM continuumio/miniconda3
```
Comment (mwalker174) [inline dockerfiles/pangenie/Dockerfile:20]: Can you peg this to a specific tag?

File: dockerfiles/pangenie/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+RUN apt-get update && apt-get install --no-install-recommends -qqy \
+    build-essential \
+    git \
+    make \
+    cmake \
+    wget \
+    pkg-config
+
+RUN git clone https://jana_ebler@bitbucket.org/jana_ebler/pangenie.git && \
+    cd pangenie && \
```
Comment (mwalker174) [inline dockerfiles/pangenie/Dockerfile:31]: Can you peg this to the current release?

File: dockerfiles/pangenie/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    conda env create -f environment.yml
+
+# Make RUN commands use the new environment:
+RUN echo "conda activate pangenie" >> ~/.bashrc
+SHELL ["/bin/bash", "--login", "-c"]
+
+# Install PanGenie
+# Without setting `PKG_CONFIG_PATH`, cmake/make cannot resolve
+# jellyfish dependency.
+RUN conda install -y jellyfish && \
```
Comment (mwalker174) [inline dockerfiles/pangenie/Dockerfile:41]: Also peg jellyfish version

**Review by mwalker174 — COMMENTED** [review]
Thank you @VJalili! I would suggest specifying specific versions for most of the installs. Also could you add a quick cleanup step at the end to delete any temp files?

File: dockerfiles/pangenie/Dockerfile (hunk 1 above)
Comment (VJalili) [reply dockerfiles/pangenie/Dockerfile]: Good point, thanks! I pinned the docker image and all the tools installed to their current latest tag and versions respectively.

**Comment by VJalili** [comment]
Thank you @mwalker174 for the suggestions.

I added a cleanup code. The image size is `1.3GB` compressed, and the cleanup code does not change it noticeably. The largest directories are those related to conda, and [it seems that can be reduced through a multi-stage build](https://pythonspeed.com/articles/conda-docker-image-size/). IMHO, this can be a good optimization when improving this Docker image for a large-scale execution.

**Review by mwalker174 — APPROVED** [review]
Thank you for incorporating my suggestions. I'd also suggest adding a build of this to `input_values/dockers.json` even though there is no WDL for this yet, just so we know where to find a build of this.

**Comment by VJalili** [comment]
Thank you for reviewing this @mwalker174.

**Pull request merged**